eglot.el

```
@@ -1735,11 +1759,7 @@ THINGS are either registrations or unregisterations (sic)."
   (append
    (eglot--VersionedTextDocumentIdentifier)
    (list :languageId
-	 (cond
-           ((get major-mode 'eglot-language-id))
-           ((string-match "\\(.*\\)-mode" (symbol-name major-mode))
-            (match-string 1 (symbol-name major-mode)))
-           (t "unknown"))
+	 (eglot--language-id (eglot-current-server))
```

I don't know if this check needs to be guarded, ie. will there always be a current server? If so, how best to handle it?

---

The guard is using `eglot--current-server-or-lose` .  We must have a language id and having nil as the server will just blow up.  let it blow up with a better error.  But it shouldn't, the server should always be bound in the relevant operations.